### PR TITLE
chore(fx-2377, fx-2371): remove old fair apps; redirects

### PR DESCRIPTION
**Pull request by dzucconi**
Closes: [FX-2377](https://artsyproduct.atlassian.net/browse/FX-2377), [FX-2371](https://artsyproduct.atlassian.net/browse/FX-2371)

Deletes the old apps, simplifies the redirects to always happen instead of using the env vars.

------

<del>I'm curious about what the `fair_info` app is — it looks like that can be removed as well (but can be addressed in a separate PR)?</del> — removed

### CODE REVIEW

File: src/desktop/apps/fair/redirectFairRequests.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    return res.redirect(301, `/fair/${fairSlug}/info`)
+  }
+
+  // this matches /:id/browse/artworks, but not /:id/browse/exhibitors
+  if (req.route.path === "/:id/browse/*" && req.params["0"] === "artworks") {
+    return res.redirect(301, `/fair/${fairSlug}/artworks`)
+  }
+
+  return res.redirect(301, basePath)
+}
```
Comment (damassi) [inline src/desktop/apps/fair/redirectFairRequests.ts:32]: Note that all new server-side code for our SPA-style apps [now lives here](https://github.com/artsy/force/blob/master/src/desktop/apps/artsy-v2/apps/). Redirects, etc, can live there; and here's a [couple](https://github.com/artsy/force/blob/master/src/desktop/apps/artsy-v2/apps/artwork/artworkMiddleware.tsx) [examples](https://github.com/artsy/force/blob/master/src/desktop/apps/artsy-v2/middleware/userRequiredMiddleware.tsx)

Comment (dzucconi) [reply src/desktop/apps/fair/redirectFairRequests.ts:32]: I very much do not wish to try to figure out how to deal with the fall through redirects for the bare profile IDs routes at this moment. It seems like it'd be easier to deal with that once we don't have any other vanity-style routes.

Comment (dzucconi) [reply src/desktop/apps/fair/redirectFairRequests.ts:32]: (But if you think it's not as painful as I'm imagining I can try tackling in a separate PR)

Comment (damassi) [reply src/desktop/apps/fair/redirectFairRequests.ts:32]: No clue how painful it is, mostly just pointing out that we've moved a lot of this kind of logic over to the v2 folder. Np leaving it here for the tiime being, but might be worth adding a note of some sort at the top mentioning that the app has been moved into our app shell (or something).

Comment (dzucconi) [reply src/desktop/apps/fair/redirectFairRequests.ts:32]: Yeah, good idea; will add a README in the app dir for now.

Comment (damassi) [reply src/desktop/apps/fair/redirectFairRequests.ts:32]: even better 👍

Comment (dzucconi) [reply src/desktop/apps/fair/redirectFairRequests.ts:32]: Oh, and I also see your original point (organization). For some reason I thought you were talking about moving the routes over to found. Will move!

Comment (dzucconi) [reply src/desktop/apps/fair/redirectFairRequests.ts]: 284b1a8dd41c419798030465446a607cc4bd18ce 🙇‍♂️

File: src/desktop/apps/fair/redirectFairRequests.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+): void => {
+  const isFairProfile = res.locals?.profile?.get("owner_type") === "Fair"
+
+  if (!isFairProfile) return next()
+
+  const fairSlug: string | null = res.locals.profile.get("owner")?.id
+
+  if (!fairSlug) {
+    return res.redirect(302, "/art-fairs")
+  }
```
Comment (anandaroop) [inline src/desktop/apps/fair/redirectFairRequests.ts]: 👍

**Review by anandaroop — APPROVED** [review]
LGTM!

One question tho — is the [README](https://github.com/artsy/force/blob/769cfa63b9840588dc97ecc24ea7d9754359d46b/src/desktop/apps/artsy-v2/apps/fair/README.md) intentionally empty?

**Comment by dzucconi** [comment]
(Whoops removed empty readme, no longer needed)

File: src/desktop/apps/fair_info/index.coffee [hunk 3]
```diff
@@ -1,12 +1,12 @@
 express = require 'express'
 routes = require './routes'
-{ redirectFairRequests } = require('../fair/fairRedirection')
+{ fairRedirectionMiddleware } = require('../artsy-v2/apps/fair/fairRedirectionMiddleware')
 
 app = module.exports = express()
 app.set 'views', __dirname + '/templates'
 app.set 'view engine', 'jade'
 
-app.use '/:id/*', redirectFairRequests, routes.assignFair
```
Comment (dblandin) [inline src/desktop/apps/fair_info/index.coffee:9]: A couple questions:

- Is the `routes.assignFair` behavior still needed if we're querying the fair via our Metaphysics GraphQL layer?
- Do we still rely on these `/info/*` routes?

File: src/desktop/assets/fairs.ts [hunk 4]
```diff
@@ -7,6 +7,4 @@ if (location.pathname === "/art-fairs") {
   $(require("../apps/fair_info/client/index.coffee").init)
```
Comment (dblandin) [inline src/desktop/assets/fairs.ts:7]: Do we still rely on the `fair_info` app?

**Review by dblandin — APPROVED** [review]
Left a couple questions about `routes.assignFair` and the `fair_info` app but overall this looks great!

File: src/desktop/apps/fair_info/index.coffee (hunk 3 above)
Comment (dzucconi) [reply src/desktop/apps/fair_info/index.coffee:9]: This is a separate app that I haven't looked at yet.

**Pull request merged**